test/index.test.ts

```diff
@@ -592,6 +594,46 @@ describe('Unit | Utility | changeset', () => {
     expect(actual).toEqual(expectedResult);
   });
 
+  it('#set works after save', () => {
+    dummyModel['org'] = {
+      usa: {
+        mn: 'mn',
+        ny: 'ny'
+      }
+    };
+
+    const c = Changeset(dummyModel);
+    c.set('org.usa.ny', 'NY');
+    c.set('org.usa.mn', 'MN');
+
+    expect(c.get('org.usa.ny')).toBe('NY');
+    expect(c.get('org.usa.mn')).toBe('MN');
+    expect(dummyModel.org.usa.ny).toBe('ny');
+    expect(dummyModel.org.usa.mn).toBe('mn');
+
+    c.save();
+
+    expect(c.get('org.usa.ny')).toBe('NY');
+    expect(c.get('org.usa.mn')).toBe('MN');
+    expect(dummyModel.org.usa.ny).toBe('NY');
+    expect(dummyModel.org.usa.mn).toBe('MN');
+
+    c.set('org.usa.ny', 'nil');
+    c.set('org.usa.mn', 'undefined');
+
+    expect(c.get('org.usa.ny')).toBe('nil');
+    expect(c.get('org.usa.mn')).toBe('undefined');
+    expect(dummyModel.org.usa.ny).toBe('NY');
+    expect(dummyModel.org.usa.mn).toBe('MN');
+
+    c.save();
+
+    expect(c.get('org.usa.ny')).toBe('nil');
+    expect(c.get('org.usa.mn')).toBe('undefined');
+    expect(dummyModel.org.usa.ny).toBe('nil');
+    expect(dummyModel.org.usa.mn).toBe('undefined');
+  });
```

Comment:
Here is a test that captures more of what was going on. Eg. setting only one property at the time.
```
  it('#set works after save', () => {
    dummyModel['org'] = {
      usa: {
        mn: 'mn',
        ny: 'ny'
      }
    };

    const c = Changeset(dummyModel);
    c.set('org.usa.ny', 'NY');
    c.set('org.usa.mn', 'MN');

    expect(c.get('org.usa.ny')).toBe('NY');
    expect(c.get('org.usa.mn')).toBe('MN');
    expect(dummyModel.org.usa.ny).toBe('ny');
    expect(dummyModel.org.usa.mn).toBe('mn');

    c.save();

    expect(c.get('org.usa.ny')).toBe('NY');
    expect(c.get('org.usa.mn')).toBe('MN');
    expect(dummyModel.org.usa.ny).toBe('NY');
    expect(dummyModel.org.usa.mn).toBe('MN');

    c.set('org.usa.ny', 'nil');

    expect(c.get('org.usa.ny')).toBe('nil');
    expect(c.get('org.usa.mn')).toBe('MN');
    expect(dummyModel.org.usa.ny).toBe('NY');
    expect(dummyModel.org.usa.mn).toBe('MN');

    c.save();

    expect(c.get('org.usa.ny')).toBe('nil');
    expect(c.get('org.usa.mn')).toBe('MN');
    expect(dummyModel.org.usa.ny).toBe('nil');
    expect(dummyModel.org.usa.mn).toBe('MN');

    c.set('org.usa.ny', 'nil2');
    c.set('org.usa.mn', 'undefined');

    expect(c.get('org.usa.ny')).toBe('nil2');
    expect(c.get('org.usa.mn')).toBe('undefined');
    expect(dummyModel.org.usa.ny).toBe('nil');
    expect(dummyModel.org.usa.mn).toBe('MN');

    c.save();

    expect(c.get('org.usa.ny')).toBe('nil2');
    expect(c.get('org.usa.mn')).toBe('undefined');
    expect(dummyModel.org.usa.ny).toBe('nil2');
    expect(dummyModel.org.usa.mn).toBe('undefined');
  });
```

Reply:
This is now fixed in #12 !!!